### PR TITLE
Fix concatenation logic: Add support for uglifyJS 'unsafe' option

### PR DESCRIPTION
**Pull request by everdimension**
Fixes https://github.com/js-cookie/js-cookie/issues/206

### CODE REVIEW

**Comment by everdimension** [comment]
Used gulp-compare-size to get this info:

   | `unsafe` option | raw size | gZipped size
----|--------------------- | ----------- | ----------------
current build | false | 1710 | 894 
current build <sup>bugged behavior</sup> | true | 1697 | 891
build from PR | false | 1716 | 895
build from PR | true | 1703 | 891

So i'd say this PR practically doesn't affect the bundle size but makes it more robust.

**Comment by FagnerMartinsBrack** [comment]
We need to run the tests in the minified file using the `unsafe` option in js-cookie build. Can you do that? If all tests pass and it spares 4 bytes as the table you posted, the this is definitely good to go!

**Comment by everdimension** [comment]
@FagnerMartinsBrack updated the PR with relevatnt changes to the tests

File: test/missing_semicolon.html [hunk 1]
```diff
@@ -21,4 +21,4 @@
 <body>
 
 </body>
-</html>
\ No newline at end of file
+</html>
```
Comment (FagnerMartinsBrack) [inline test/missing_semicolon.html]: Can we please remove this unrelated change so that the changes are logged only in the relevant lines? Thanks!

Comment (everdimension) [inline test/missing_semicolon.html]: Right, sorry. I'm using atom which does this by default. I'll undo this and squash the commit.

Comment (everdimension) [inline test/missing_semicolon.html]: Done ;)
